scripts/_functions_test_v1.py

```diff
@@ -0,0 +1,248 @@
+        username = auth['cloudfunctions_username']
+        password = auth['cloudfunctions_password']
+
+    try:
+        inputFile = open(args.inputFileName, 'r')
+    except IOError:
+        logger.critical('Cannot open test input file %s', args.inputFileName)
+        sys.exit(1)
+
+    try:
+        outputFile = open(args.outputFileName, 'w')
+    except IOError:
+        logger.critical('Cannot open test output file %s', args.outputFileName)
+        sys.exit(1)
+
+    try:
+        inputJson = json.load(inputFile)
+    except ValueError as e:
+        logger.critical('Cannot decode json from test input file %s, error: %s', args.inputFileName, str(e))
+        sys.exit(1)
+
+    if not isinstance(inputJson, list):
+        logger.critical('Input test json is not array!')
+        sys.exit(1)
+
+    replaceDict = {}
+    for attr in dir(config):
+        if not attr.startswith("__"):
+            if attr == 'replace':
+                # format \'valueToBeReplaced1:replacement1,valueToBeReplaced2:replacement2\'
+                replacementsString = getattr(config, attr)
+                for replacementString in replacementsString.split(','):
+                    replacementStringSplit = replacementString.split(':')
+                    if len(replacementStringSplit) != 2 or not replacementStringSplit[0] or not replacementStringSplit[1]:
+                        logger.critical('Invalid format of \'replace\' parameter, valid format is \'valueToBeReplaced1:replacement1,valueToBeReplaced2:replacement2\'')
+                        sys.exit(1)
+                    replaceDict[replacementStringSplit[0]] = replacementStringSplit[1]
+            else:
+                replaceDict[attr] = getattr(config, attr)
+
+    # run tests
+    testCounter = 0
+    for test in inputJson:
+        if not isinstance(test, dict):
+            logger.error('Input test array element %d is not dictionary. Each test has to be dictionary, please see doc!', testCounter)
+            continue
+        logger.info('Test number: %d, name: %s', testCounter, (test['name'] if 'name' in test else '-'))
+
+        # load test input payload json
+        testInputJson = test['input']
+        testInputPath = None
+        try:
+            if testInputJson.startswith('@'):
+                testInputPath = os.path.join(os.path.dirname(args.inputFileName), testInputJson[1:])
+                logger.debug('Loading input payload from file: %s', testInputPath)
+                try:
+                    inputFile = open(testInputPath, 'r')
+                except IOError:
+                    logger.error('Cannot open input payload from file: %s', testInputPath)
+                    continue
+                try:
+                    testInputJson = json.load(inputFile)
+                except ValueError as e:
+                    logger.error('Cannot decode json from input payload from file %s, error: %s', testInputPath, str(e))
+                    continue
+        except:
```

Comment:
It might be a good practice to catch only the expected exception :). In this case probably AttributeError.
```suggestion
        except AttributeError:
```